src/reducers/tokens/index.js

```diff
@@ -0,0 +1,68 @@
+import { handleActions } from 'redux-actions'
+import reduceReducers from 'reduce-reducers'
+
+import { tokens } from '../../actions/tokens'
+
+const initialState = {
+    likelyContracts: [],
+    tokens: {}
+}
+
+const likelyContractsReducer = handleActions({
+    [tokens.likelyContracts.get]: (state, { ready, error, payload }) =>
+        (!ready || error)
+            ? state
+            : ({
+                ...state,
+                likelyContracts: payload
+            }),
+}, initialState)
+
+const tokensReducer = handleActions({
+    [tokens.tokensDetails.set]: (state, { payload }) => ({
```

Comment:
feel like we should have some better handling here than current one (replacing whole list)

like if we start from cached list – we probably only want to invalidate balances for existing contracts, but keep cached metadata as is until latest version is loaded

Reply:
done